pkg/ext-proc/backend/pod_client.go

```diff
@@ -0,0 +1,32 @@
+package backend
+
+import (
+	"fmt"
+	"net/http"
+
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/expfmt"
+	klog "k8s.io/klog/v2"
+)
+
+type PodMetricsClientImpl struct {
+}
+
+// FetchMetrics fetches metrics from a given pod.
+func (p *PodMetricsClientImpl) FetchMetrics(pod Pod) (map[string]*dto.MetricFamily, error) {
+	url := fmt.Sprintf("http://%s/metrics", pod.Address)
```

Comment:
This seems to assume that metrics are exposed at the same address as the target pod under the /metrics path. Shouldn’t we consider making this configurable in the future?

Reply:
Good point! I added a TODO for now to consider making this configurable.

Reply:
We may need to have this configured on the LLMServerPool api?

Reply:
Eventually yes I think. Now we captured a new field that can go to the LLMServerPool. Will create an issue to track this.

Reply:
https://github.com/kubernetes-sigs/llm-instance-gateway/issues/16